src/lib/dialog/dialog-directives.ts

```diff
@@ -0,0 +1,49 @@
+import {Directive} from '@angular/core';
+import {MdDialog} from './dialog';
+
+
+/**
+ * Button that will close the current dialog.
+ */
+@Directive({
+  selector: 'button[md-dialog-close]',
+  host: {
+    '(click)': 'dialog.closeTop()'
```

Comment:
Was going to say pretty much the same thing.  As is it will break some more exotic use cases.